backend/probloom/prob/views.py

```diff
@@ -128,6 +127,94 @@ def userStatistics(request, id=0):
         )
 
 
+class ProblemSetListView(View):
+    def get(self, request: HttpRequest, **kwargs) -> HttpResponse:
+        if not request.user.is_authenticated:
+            return HttpResponse(status=401)
```

Comment:
If you are using class-based views, you can use [`login_required`](https://docs.djangoproject.com/en/3.2/topics/auth/default/#the-login-required-decorator) decorator with [`method_decorator`](https://docs.djangoproject.com/en/3.2/topics/class-based-views/intro/#decorating-the-class) or [`LoginRequiredMixin`](https://docs.djangoproject.com/en/3.2/topics/auth/default/#the-loginrequired-mixin). I will make our specification for authentication different from that of HW4 so that we can use `login_required` and `LoginRequiredMixin`.